adapters/resetdigital/resetdigital.go

```diff
@@ -0,0 +1,284 @@
+			"Content-Type":    "application/json;charset=utf-8",
+			"Accept":          "application/json",
+		})
+	}
+
+	return headers
+}
+func addNonEmptyHeaders(headers *http.Header, headerValues map[string]string) {
+	for key, value := range headerValues {
+		if len(value) > 0 {
+			headers.Add(key, value)
+		}
+	}
+}
+
+func getReferer(request *openrtb2.BidRequest) string {
+	if request.Site == nil {
+		return ""
+	}
+
+	return request.Site.Domain
+}
+
+func getCurrency(request *openrtb2.BidRequest) string {
+	if len(request.Cur) == 0 {
+		return "USD"
+	}
+
+	return request.Cur[0]
+}
+
+func (a *adapter) MakeRequests(requestData *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var (
+		requests []*adapters.RequestData
+		errors   []error
+	)
+
+	for i := range requestData.Imp {
+		imp := requestData.Imp[i]
+		bidType, err := getBidType(imp)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		splittedRequestData, err := processDataFromRequest(requestData, imp, bidType)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requestBody, err := json.Marshal(splittedRequestData)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requests = append(requests, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpointUri,
+			Body:    requestBody,
+			Headers: getHeaders(requestData),
+		})
+	}
+
+	return requests, errors
+}
+
+func processDataFromRequest(requestData *openrtb2.BidRequest, imp openrtb2.Imp, bidType openrtb_ext.BidType) (resetDigitalRequest, error) {
+	var resetDigitalRequestData resetDigitalRequest
+
+	// Check if requestData.Site is not nil before accessing its fields
+	if requestData.Site != nil {
+		resetDigitalRequestData.Site.Domain = requestData.Site.Domain
+		resetDigitalRequestData.Site.Referrer = requestData.Site.Page
+	}
+
+	resetDigitalRequestData.Imps = append(resetDigitalRequestData.Imps, resetDigitalRequestImps{})
+	resetDigitalRequestData.Imps[0].BidID = requestData.ID
+	resetDigitalRequestData.Imps[0].ImpID = imp.ID
+
+	if bidType == openrtb_ext.BidTypeBanner {
+		resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes = append(resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes, []int64{imp.Banner.Format[0].W, imp.Banner.Format[0].H})
+	}
+	if bidType == openrtb_ext.BidTypeVideo {
+		resetDigitalRequestData.Imps[0].MediaTypes.Video.Sizes = append(resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes, []int64{*imp.Video.W, *imp.Video.H})
+	}
+
+	var extData = make(map[string]interface{})
+	err := json.Unmarshal(imp.Ext, &extData)
+	if err != nil {
+		return resetDigitalRequestData, err
+	}
+
+	resetDigitalRequestData.Imps[0].ZoneID.PlacementID = extData["bidder"].(map[string]interface{})["placement_id"].(string)
+	if resetDigitalRequestData.Imps[0].ZoneID.PlacementID == "test" {
+		resetDigitalRequestData.Imps[0].ForceBid = true
+	}
+
+	return resetDigitalRequestData, nil
```

Comment:
Current code can result in runtime panic if type casting for bidder `extData["bidder"].(map[string]interface{}` and placement id `["placement_id"].(string)` fails. 

Instead, will recommend making use structs. 

1. Add `ExtImpResetDigital` struct. This will be similar to `ExtImpAppNexus`.  ExtImpResetDigital can have PlacementID field.
  https://github.com/prebid/prebid-server/blob/ec729e6e5d50a1306d76332fda52108b9d313c1d/openrtb_ext/imp_appnexus.go#L12
2.  update adapter code to unmarshal `imp.Ext` into `adapters.ExtImpBidder` struct
3.  resulting value of step 2 can be unmarshalled into `ExtImpResetDigital` struct

Refer following as example,

```
openrtb_ext/imp_resetdigital.go
   // ExtImpResetDigital defines the contract for bidrequest.imp[i].ext.prebid.bidder.resetdigital
   type ExtImpResetDigital struct {
       PlacementID string  `json:"placementId"`
   }


adapters/resetdigital/resetdigital.go
	var bidderExt adapters.ExtImpBidder
	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
		return nil, err
	}

       	var resetDigitalExt openrtb_ext.ExtImpResetDigital
	if err := json.Unmarshal(bidderExt.Bidder, &appnexusExt); err != nil {
		return nil, err
	}

      resetDigitalRequestData.Imps[0].ZoneID.PlacementID = resetDigitalExt.PlacementID
      ...       
```